src/main/java/dev/morling/onebrc/CalculateAverage_yavuztas.java

```diff
@@ -0,0 +1,289 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+public class CalculateAverage_yavuztas {
+
+    private static final Path FILE = Path.of("./measurements.txt");
+
+    static class Measurement {
+        private double min;
+        private double max;
+        private double sum;
+        private int count = 1;
+
+        public Measurement(double initial) {
+            this.min = initial;
+            this.max = initial;
+            this.sum = initial;
+        }
+
+        public String toString() {
+            return round(this.min) + "/" + round(this.sum / this.count) + "/" + round(this.max);
+        }
+
+        private double round(double value) {
+            return Math.round(value * 10.0) / 10.0;
+        }
+    }
+
+    static class KeyBuffer {
+
+        ByteBuffer value;
+        int hash;
+
+        public KeyBuffer(ByteBuffer buffer) {
+            this.value = buffer;
+            this.hash = buffer.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+
+            final KeyBuffer keyBuffer = (KeyBuffer) o;
+            if (o == null || getClass() != o.getClass() || this.hash != keyBuffer.hash)
+                return false;
+
+            return this.value.equals(keyBuffer.value);
+        }
+
+        @Override
+        public int hashCode() {
+            return this.hash;
+        }
+
+        @Override
+        public String toString() {
+            final int limit = this.value.limit();
+            final byte[] bytes = new byte[limit];
+            this.value.get(bytes);
+            return new String(bytes, 0, limit, StandardCharsets.UTF_8);
+        }
+    }
+
+    static class FixedRegionDataAccessor {
+
+        static final byte SEMI_COLON = 59; // ';'
+        static final byte LINE_BREAK = 10; // '\n'
+
+        final byte[] workBuffer = new byte[64]; // assuming max 64 bytes for a row is enough
```

Comment:
station ids can be 100 char long as per the requirements. Should add a test for this, yeah.

Reply:
Oh, didn't know. It's increased now. Hopefully, 256 bytes is enough. Anyway, it doesn't hurt so much since it's only created once for each segment and then reused.